docs/development/design/build-images.rst

```diff
@@ -51,88 +51,101 @@ Goals
 
 
 New build image structure
 -------------------------
 
-.. Taken from https://github.com/readthedocs/readthedocs-docker-images/blob/master/Dockerfile
+.. Some of this is borrowed from CircleCI image versioning
 
-* ``ubuntu20-base``
-  * labels
-  * environment variables
-  * system dependencies
-  * install requirements
-  * user requirements
-    * plantuml, imagemagick, rsgv-convert, swig
-    * sphinx-js dependencies
-    * rust
-  * UID and GID
+``readthedocs/build:base``
+    Alias of ``readthedocs/build:base-ubuntu20``, used to abstract concept of OS
+    away from end users where it's not neccessary.
 
-* ``ubuntu20-pdf`` (from ``ubuntu20-base``)
-  * PDF/LaTeX dependencies
+    Base images also include:
 
-* ``ubuntu20`` (from ``ubuntu20-pdf``)
-  * all Python versions (2, 3.6, 3.7, 3.8, 3.9)
-  * conda
-  * future extra user requirements
-  * labels
+    ``readthedocs/build:base-ubuntu20``
+        Base for next gen prod images. Synonymous with current concept of ``latest``
 
-We will also build a ``nopdf`` version to allow quick testing in local development:
+    ``readthedocs/build:base-ubuntu18``
+        Base for previous gen prod images. Synonymous with concept of ``stable``
 
-* ``ubuntu20-nopdf`` (from ``ubuntu20-base``)
-  * same as ``ubuntu20`` but based on ``ubuntu20-base`` instead
+    Base images include:
 
-.. note::
+    * labels
+    * environment variables
+    * system dependencies
+    * install requirements
+    * PDF/LaTeX dependencies
+    * Limited user requirements
+      * plantuml, imagemagick, rsgv-convert, swig
+      * sphinx-js dependencies
+    * UID and GID
 
-   I don't think it's useful to have ``ubuntu20-py37`` exposed to users,
-   since the Python version is selected by using the config file's ``python.version`` keyword,
-   we only update patch versions and we don't remove them (unless together with OS changes).
+``readthedocs/build:python``
+    Installs latest release of Python
 
-.. Build all these images with Docker
-   docker build -t readthedocs/build:ubuntu20-base -f Dockerfile.base .
-   docker build -t readthedocs/build:ubuntu20-nopdf -f Dockerfile.nopdf .
-   docker build -t readthedocs/build:ubuntu20-pdf -f Dockerfile.pdf .
-   docker build -t readthedocs/build:ubuntu20 -f Dockerfile .
+``readthedocs/build:python39``
+    Installs 3.9 release of Python
 
-   Check the shared space between images
-   docker system df --verbose | grep -E 'SHARED SIZE|readthedocs'
+``readthedocs/build:python39-ubuntu18``
+    Installs 3.9 release of Python on Ubuntu 18.04 base
+    (``readthedocs/build:base-ubuntu18``)
 
+``readthedocs/build:python27``
+    Installs 3.9 release of Python
 
-Custom images
--------------
+``readthedocs/build:conda12``
+    Installs 1.2 release of Conda
+
+``readthedocs/build:node12``
+    Installs Node 12.x release
+
+``readthedocs/build:python+node``
+    Installs latest Python and latest Node
```

Comment:
This is where things get screwy. I'm not a huge fan, but this seems better than arbitrarily selecting a version of node or treating all languages that != python as second class. Python is the big one for us though of course.